tests/monotouch-test/Foundation/DateTest.cs

```diff
@@ -72,8 +71,20 @@ public void Precision32022 ()
 		{
 			var a = NSDate.Now;
 			var b = a.SecondsSinceReferenceDate - ((NSDate) (DateTime) a).SecondsSinceReferenceDate;
-			// ensure decimals are not truncated - but there's an unavoidable loss of precision
-			Assert.AreEqual (b, 0, 0.00001, "1");
+
+			// ensure decimals are not truncated - but there's an unavoidable loss of precision from ns to ms
+			Assert.AreEqual (b, 0, 0.001, "1");
```

Comment:
you know why the precision changed ?
or how to keep the one we already have ?

Reply:
The original precision came from using the `SecondsSinceReferenceDate` and storing the value as a double.

With the new method of constructing from components/calendar we end up converting between nanoseconds and milliseconds, since these are the smallest units `DateTime` and `NSDate` use. Milliseconds reduce precision to `0.001`